lang.cpp

```diff
@@ -27,68 +35,111 @@ bool isSeparator(char list){
   }
   return false;
 }
-string keywords[] = {"close", "if", "else", "elseif"};
 
+string keywords[] = {"if", "else", "elseif"};
+bool isKeyword(string word){
+  for(int i = 0; i < sizeof(keywords)/sizeof(char); i++) {
+    if (word == keywords[i]){
+      return true;
+    }
+  }
+  return false;
+}
+
+bool isLiteral(string word){
+  int isDigit = 0;
+  int j = 0;
+  while((j < word.length()) && (isDigit == 0)){
+    isDigit = isdigit(word[j]);
+    j++;
+  }
+  return isDigit;
+}
+
+void push(string word){
+  if(word != ""){
+    tmp_list.push_back(word);
+    if(isKeyword(word)){
+      tmp_tokens.push_back("Keyword");
+     }else if(isLiteral(word)){
+       tmp_tokens.push_back("Literal");
+    }else{
+      tmp_tokens.push_back("Identifier");
+    }
+  }
+}
 
 int main(){
-  ifstream code("1.txt");
-  vector<string> code_list;
-  string word = "";
+  FILE *code;
+  if((code = fopen("1.txt", "r")) == NULL) {
+    printf ("Cannot open file.\n");
+    return 1;
+  }
 
   char ch;
-  while (!code.eof()) {
-      code >> ch;
-      if((isOperator(ch)) || (isSeparator(ch))){
-        code_list.push_back(word);
-        word = ch;
-        code_list.push_back(word);
-        word = "";
-      }else{
-        word += ch;
-      }
+  string word = "";
+  while((ch = getc(code)) != EOF){
```

Comment:
You need to add a variable reset in the function, since this action is repeated many times